SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp

```diff
@@ -0,0 +1,271 @@
+//===-- ctor_load_acc.hpp - Generic code for tests on simd load constructors
+//      from accessor -----------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd load constructor from
+/// accessor
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+#define ESIMD_TESTS_DISABLE_DEPRECATED_TEST_DESCRIPTION_FOR_LOGS
+
+#include "../sycl_accessor.hpp"
+#include "../sycl_range.hpp"
+#include "ctor_load.hpp"
+
+#include <cassert>
+#include <cstring>
+#include <memory>
+#include <sstream>
+
+namespace esimd_test::api::functional::ctors {
+
+// Utility class for offset generation based on alignment and multiplicator
+// given
+// Offset should be a multiplier of alignment for simd load constructor to work
+// as expected
```

Comment:
There should be a dependency on `alignof(DataT)` also due to such restriction for simd load constructor